### PR TITLE
Fix concurrent accepts

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: cli/resources.rs [hunk 1]
```diff
@@ -171,12 +171,50 @@ impl Resource {
...
+  }
+
   // close(2) is done by dropping the value. Therefore we just need to remove
   // the resource from the RESOURCE_TABLE.
   pub fn close(&self) {
     let mut table = RESOURCE_TABLE.lock().unwrap();
     let r = table.remove(&self.rid);
     assert!(r.is_some());
+    // If TcpListener, we must kill all pending accepts!
+    if let Repr::TcpListener(_, Some(t)) = r.unwrap() {
```
Comment (piscisaureus) [inline cli/resources.rs]: Maybe you can move the unwrap two lines up and drop the assert?

Comment (ry) [reply cli/resources.rs]: done

**Comment by piscisaureus** [comment]
I don't really understand how this works and I'd like to...
How does storing / forgetting the associated task cancel a future?
But it seems to pass the test so I'll approve it.

**Comment by ry** [comment]
@piscisaureus When a TCPListener is closed, if there are pending accept futures for that server, they will not be woken up. (This seems like a bug in Tokio, actually.) The solution is to associate the Accept future (actually the current task) with the TCPListener and manually poll it (using Task::notify)

**Comment by zekth** [comment]
Is an issue created in Tokio? might be a good addition to the code if so. (to explain)

**Pull request merged**